src/engine/QueryExecutionContext.h

```diff
@@ -116,6 +118,17 @@ class QueryExecutionContext {
     updateCallback_(nlohmann::ordered_json(runtimeInformation).dump());
   }
 
+  // Getter for the global `NEXT_NEWBLANKNODEINDEX` counter.
+  [[nodiscard]] static int getNextNewBlankNodeIndex() {
```

Comment:
Don't make this function `static`, even though it can currently be. The staticness should in this case be an implementation detail.